### ISSUE TITLE
Use the MEGA65's built-in 3.5" drive as C1581

### ISSUE THREAD

**Issue by sy2002** [MJoergen/C64MEGA65#106]
MiSTer offers simulation code for the C1581 that could be used to catch two birds with one stone:

1) Simulate a C1581 and use the SD card to load/save data and use the M2M framework's VDrive system (that should work out of the box with the C1581). <== This is another topic. It is on our ROADMAP.md

2) Use the MiSter code to generate control signals for a to-be-done controller for the 3.5" floppy (which is a PC drive) and therefore have a very compatible C1581.

Basically there are two ways to use the MEGA65's built-in 3.5" drive as a C1581:

a) Write a controller that wraps the different control signals of the drive (it is a IBM PC drive) and translates them into the IEC bus protocol. This would not need MiSTer's simulated drive at all. We would "just" have an additional "participant" at the IEC bus. From a compatibility standpoint: This would work with all standard applications and is bascially what an SD2IEC does with an SD card.

b) MiSTer's simulation of the C1581 might be generating signals to move the read/write head, to drive the motor etc. So in a move to be even more compatible than (a), we could use the signals that MiSTer's simulation generates and all the encoding, etc.

This is a task that needs research and some decisions before we can continue.

**Comment by sy2002** [comment]
@MJoergen Question from deft: Do you want to try to tackle this for V6? If so, we would be "complete" in terms of supporting the whole MEGA65 machine in the C64 for MEGA65 core. In this case, please tag as V6. But totally OK, if not.

**Comment by MJoergen** [comment]
No, this will not be part of V6. The development effort is large, and will need a lot of testing to ensure compatibility. This is better suited for a later release.

**Comment by sy2002** [comment]
Implemented in Alpha 18X2
